zb-db/src/test/java/io/camunda/zeebe/db/impl/ColumnFamilyPropertyTest.java

```diff
@@ -0,0 +1,205 @@
+    operations.forEach(op -> op.runOnColumnFamily(columnFamily));
+    columnFamilyBehavedLikeMap(columnFamily, map);
+  }
+
+  private void columnFamilyBehavedLikeMap(
+      final ColumnFamily<DbLong, DbLong> columnFamily, final Map<Long, Long> map) {
+    map.forEach(
+        (key, value) -> {
+          final var dbKey = new DbLong();
+          dbKey.wrapLong(key);
+          assertThat(columnFamily.get(dbKey).getValue())
+              .as("Key " + dbKey.getValue() + " should have value " + value)
+              .isEqualTo(value);
+        });
+    columnFamily.forEach(
+        (key, value) -> assertThat(map).containsEntry(key.getValue(), value.getValue()));
+  }
+
+  @Provide
+  ListArbitrary<TestableOperation> operations() {
+    final var k = Arbitraries.longs().greaterOrEqual(0);
+    final var v = Arbitraries.longs();
+    final var insert =
+        Combinators.combine(k, v)
+            .as(
+                (arbitraryK, arbitraryV) ->
+                    (TestableOperation) new InsertOp(arbitraryK, arbitraryV));
+    final var update =
+        Combinators.combine(k, v)
+            .as(
+                (arbitraryK, arbitraryV) ->
+                    (TestableOperation) new UpdateOp(arbitraryK, arbitraryV));
+    final var upsert =
+        Combinators.combine(k, v)
+            .as(
+                (arbitraryK, arbitraryV) ->
+                    (TestableOperation) new UpsertOp(arbitraryK, arbitraryV));
+    final var deleteExisting =
+        Combinators.combine(k, v)
+            .as(
+                (arbitraryK, arbitraryV) ->
+                    (TestableOperation) new DeleteExistingOp(arbitraryK, arbitraryV));
+    final var c =
+        Combinators.combine(k, v)
+            .as(
+                (arbitraryK, arbitraryV) ->
+                    (TestableOperation) new DeleteIfExistsOp(arbitraryK, arbitraryV));
+
+    return Arbitraries.oneOf(List.of(insert, update, upsert, deleteExisting)).list();
+  }
+
+  @BeforeProperty
+  private void setup() {
+    final File pathName = Files.newTemporaryFolder();
+    final ZeebeDbFactory<DefaultColumnFamily> dbFactory = DefaultZeebeDbFactory.getDefaultFactory();
+    final ZeebeDb<DefaultColumnFamily> zeebeDb = dbFactory.createDb(pathName);
+
+    columnFamily =
+        zeebeDb.createColumnFamily(
+            DefaultColumnFamily.DEFAULT, zeebeDb.createContext(), new DbLong(), new DbLong());
+    map = new HashMap<>();
+  }
+
+  @AfterTry
+  private void cleanup() {
+    columnFamily.forEach((k, v) -> columnFamily.delete(k));
+    map.clear();
+  }
+
+  record InsertOp(long k, long v) implements TestableOperation {
+
+    @Override
+    public void runOnColumnFamily(final ColumnFamily<DbLong, DbLong> columnFamily) {
+      final var dbKey = new DbLong();
+      final var dbValue = new DbLong();
+      dbKey.wrapLong(k);
+      dbValue.wrapLong(v);
+      try {
+        columnFamily.insert(dbKey, dbValue);
+      } catch (final Exception ignored) {
```

Comment:
```suggestion
      } catch (final ZeebeDbInconsistentException ignored) {
```

Reply:
That doesn't work, unfortunately. `TransactionContext#runInTransaction` wraps all exception in a runtime exception. But I can assert that the *cause* of the runtime exception is `ZeebeDbInconsistentException`.